src/language-js/printer-estree.js

```diff
@@ -2054,6 +2055,111 @@ function printPathNoParens(path, options, print, args) {
       return join(literalline, n.value.raw.split(/\r?\n/g));
     case "TemplateLiteral": {
       const expressions = path.map(print, "expressions");
+      const parentNode = path.getParentNode();
+
+      /**
+       * describe.each`table`(name, fn)
+       * describe.only.each`table`(name, fn)
+       * describe.skip.each`table`(name, fn)
+       * test.each`table`(name, fn)
+       * test.only.each`table`(name, fn)
+       * test.skip.each`table`(name, fn)
+       *
+       * Ref: https://github.com/facebook/jest/pull/6102
+       */
+      const jestEachTriggerRegex = /^[xf]?(describe|it|test)$/;
+      if (
+        parentNode.type === "TaggedTemplateExpression" &&
+        parentNode.quasi === n &&
+        parentNode.tag.type === "MemberExpression" &&
+        parentNode.tag.property.type === "Identifier" &&
+        parentNode.tag.property.name === "each" &&
+        ((parentNode.tag.object.type === "Identifier" &&
+          jestEachTriggerRegex.test(parentNode.tag.object.name)) ||
+          (parentNode.tag.object.type === "MemberExpression" &&
+            parentNode.tag.object.property.type === "Identifier" &&
+            (parentNode.tag.object.property.name === "only" ||
+              parentNode.tag.object.property.name === "skip") &&
+            parentNode.tag.object.object.type === "Identifier" &&
+            jestEachTriggerRegex.test(parentNode.tag.object.object.name)))
+      ) {
+        /**
+         * a    | b    | expected
+         * ${1} | ${1} | ${2}
+         * ${1} | ${2} | ${3}
+         * ${2} | ${1} | ${3}
+         */
+        const headerColumnNames = n.quasis[0].value.raw
+          .trim()
+          .split(/\s*\|\s*/);
+        if (
+          headerColumnNames.length > 1 ||
+          headerColumnNames.some(column => column.length !== 0)
+        ) {
+          const oneLineExpressions = expressions
+            .map(removeLineBreaks)
+            .map(doc => "${" + printDocToString(doc, options).formatted + "}");
+
+          const columnContents = [[]];
+          for (let i = 1; i < n.quasis.length; i++) {
+            const rowContents = columnContents[columnContents.length - 1];
+            const correspondingExpression = oneLineExpressions[i - 1];
```

Comment:
You can’t expect the input to be well formed. You need to deal with malformed ones by not triggering undefined exceptions within prettier

Reply:
There should be always `n` quasis and `n - 1` expressions in a `TemplateLiteral`, I'm not sure if it's possible to be malformed..

- `xy` -> `{ quasis: ["xy"], expressions: [] }`
- `${1}` -> `{ quasis: ["", ""], expressions: [1] }`
- `x${1}y` -> `{ quasis: ["x", "y"], expressions: [1] }`
- `${1}${2}` -> `{ quasis: ["", "", ""], expressions: [1, 2] }`

Reply:
/facepalm

You are right! Doing code review after 12 hours of flight and a huge jetlag is not a good idea! Sorry